File: EngagementHub/TOS.EngagementHub.Data/Repositories/Implementations/SkillRepository.cs
using TOS.Data.MongoDB;
using TOS.Data.MongoDB.Repositories;
using TOS.EngagementHub.Models;

namespace TOS.EngagementHub.Data.Repositories.Implementations
{
    public class SkillRepository : Repository<Skill>, ISkillRepository
    {
        public SkillRepository(IDatabaseProvider databaseProvider) : base(databaseProvider)
        {
        }
    }
}

File: NiceReads/Backend/Server/tos-server/src/application/data/repositories/SkillRepository.ts
import { inject, injectable } from 'inversify';
import { constr } from '../../../common/common.types';
import { Uuid } from '../../../common/uuid';
import { Repository } from '../../../data/repository';
import { IDatabase } from '../../../database/database';
import { Types } from '../../../types';
import { Skill } from '../../models/Skill';
import { ISkillRepository } from './ISkillRepository';

@injectable()
export class SkillRepository extends Repository<Skill, Uuid> implements ISkillRepository {
    constructor(@inject(Types.DATABASE) database: IDatabase<Uuid>) {
        super(database);
    }
    getEntityType(): constr<Skill> {
        return Skill;
    }
}